android/modules/android/src/java/ti/modules/titanium/android/AndroidModule.java

```diff
@@ -705,8 +705,7 @@ public boolean hasPermission(Object permissionObject)
 				if (filteredPermissions.size() > 0) {
 					TiBaseActivity.registerPermissionRequestCallback(REQUEST_CODE, permissionCallback,
 						callbackThisObject, promise);
-					currentActivity.requestPermissions(filteredPermissions.toArray(
-						new String[filteredPermissions.size()]), REQUEST_CODE);
+					currentActivity.requestPermissions(filteredPermissions.toArray(new String[0]), REQUEST_CODE);
```

Comment:
I was always used to pre-sizing the arrays, but after some Googling it turns out the JDKs have been optimized to handle zero-length arrays as fast or faster now, so was unaware that the ugly boilerplate code is no longer necessary!

Reply:
Yeah, the old approach used to be optimal (only does 1 allocation and uses the array given) but this approach is now the optimal way to do it and simpler to write.